src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -476,6 +481,10 @@ public ValueTask<int> SendReplyAsync(PayloadData payload, IOBehavior ioBehavior,
 				VerifyConnected();
 				task = m_payloadHandler.WritePayloadAsync(payload.ArraySegment, ioBehavior);
 			}
+			catch (Exception ex) when (m_timedOut)
+			{
+				task = ValueTaskExtensions.FromException<int>(new MySqlException("The Command Timeout was exceeded.", ex));
```

Comment:
Avoid duplication of exception message? An internal `MySqlException.TimeoutMessage` or something?

Reply:
More important to avoid duplication of the exception handling logic, I think. But first I need a PR that passes the tests, then I can clean up the code.